### PR TITLE
add output functions for objects

### PR DESCRIPTION
**Pull request by HeikoSchuett**
I think it is very helpful to define the __str__ and __repr__ functions for our objects to give a more helpful output when you type dataset or rdm in your terminal. 

Here I will implement these for dataset & RDM. If you think other information should be displayed or a different format should be used please say so!

### CODE REVIEW

File: pyrsa/data/dataset.py [hunk 1]
```diff
@@ -32,8 +33,8 @@ class DatasetBase:
         Returns:
             dataset object
     """
-    def __init__(self, measurements=None, descriptors=None,
-                 obs_descriptors=None, channel_descriptors=None):
+    def __init__(self, measurements, descriptors={},
+                 obs_descriptors={}, channel_descriptors={}):
```
Comment (JasperVanDenBosch) [inline pyrsa/data/dataset.py]: Using a mutable object like a `dict` or `list` for an argument default value is generally a bad idea, as the changed object will later server as the default again..

File: pyrsa/data/dataset.py [hunk 2]
```diff
@@ -49,6 +50,35 @@ def __init__(self, measurements=None, descriptors=None,
...
+        """
+        defines string which is printed for the object
+        """
+        return (f'pyrsa.data.{self.__class__.__name__}(\n'
+                f'measurements = \n{self.measurements}\n'
+                f'descriptors = \n{self.descriptors}\n'
+                f'obs_descriptors = \n{self.obs_descriptors}\n'
+                f'channel_descriptors = \n{self.channel_descriptors}\n'
+                )
+
```
Comment (JasperVanDenBosch) [inline pyrsa/data/dataset.py:62]: This works, another approach would be a triple-quoted f string:

```
f"""
pyrsa.data.{self.__class__.__name__}(
    measurements = {self.measurements}
    descriptors = {self.descriptors}
    obs_descriptors = {self.obs_descriptors}
    channel_descriptors = {self.channel_descriptors}
)
"""
```

**Review by JasperVanDenBosch — CHANGES_REQUESTED** [review]
Good work, this will definitely make it more user friendly. I'd recommend against the mutable defaults, otherwise looks good. I'll contribute a test

File: pyrsa/util/descriptor_utils.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+        Returns:
+            string_descriptors(String): formated string to show dict
+    """
+    string_descriptors = ''
+    for entry in descriptors:
+        string_descriptors = (string_descriptors +
+                              f'{entry} = {descriptors[entry]}\n'
+                              )
+    return string_descriptors
```
Comment (JasperVanDenBosch) [inline pyrsa/util/descriptor_utils.py:23]: how about `return ''.join([f'{k} = {v}\n' for k, v in descriptors.items()])`? Too terse perhaps?

File: pyrsa/data/dataset.py (hunk 1 above)
Comment (HeikoSchuett) [reply pyrsa/data/dataset.py]: That is definitely good to know. I did not expect that behavior here. Fixed in a new commit.

**Review by JasperVanDenBosch — APPROVED** [review]

File: pyrsa/util/descriptor_utils.py (hunk 3 above)
Comment (HeikoSchuett) [reply pyrsa/util/descriptor_utils.py:23]: well, is it faster or in some other way better than the loop? If so, I think it is hidden enough that we could change to your one-line solution; it is hidden enough after all.

File: pyrsa/data/dataset.py (hunk 2 above)
Comment (HeikoSchuett) [reply pyrsa/data/dataset.py:62]: True, in this case I like the one string per output version. Do you have a strong preference for the triple-quoted string?

File: pyrsa/util/descriptor_utils.py (hunk 3 above)
Comment (JasperVanDenBosch) [reply pyrsa/util/descriptor_utils.py:23]: No both are fine

File: pyrsa/data/dataset.py (hunk 2 above)
Comment (JasperVanDenBosch) [reply pyrsa/data/dataset.py:62]: nope, no preference

**Review by g14r — APPROVED** [review]

**Pull request merged**